### PR TITLE
feat(rds): dual-stack mode support

### PR DESCRIPTION
**Pull request by Tietew**
This PR adds dual-stack mode support to RDS instances and clusters.

### Aurora
- Working with a DB cluster in a VPC  
  https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/USER_VPC.WorkingWithRDSInstanceinaVPC.html
- CloudFormation AWS::RDS::DBCluster  
  https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-rds-dbcluster.html#cfn-rds-dbcluster-networktype

```ts
declare const vpc: ec2.Vpc; // VPC and subnets must have IPv6 CIDR blocks
const cluster = new rds.DatabaseCluster(this, 'Database', {
  engine: rds.DatabaseClusterEngine.auroraMysql({ version: rds.AuroraMysqlEngineVersion.VER_3_02_1 }),
  instanceProps: {
    vpc,
    publiclyAccessible: false,
  },
  networkType: rds.NetworkType.DUAL,
});
```

### RDS
- Working with a DB instance in a VPC  
  https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_VPC.WorkingWithRDSInstanceinaVPC.html
- CloudFormation AWS::RDS::DBInstance  
  https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-rds-dbinstance.html#cfn-rds-dbinstance-networktype

```ts
declare const vpc: ec2.Vpc; // VPC and subnets must have IPv6 CIDR blocks
const instance = new rds.DatabaseInstance(this, 'Instance', {
  engine: rds.DatabaseInstanceEngine.postgres({ version: rds.PostgresEngineVersion.VER_14_4 }),
  vpc,
  networkType: rds.NetworkType.DUAL,
  publiclyAccessible: false,
});
```

Note: CDK cannot check whether the specified VPC and subnets have actually IPv6 CIDR blocks because `ec2.IVpc` and `ec2.ISubnet` does not have ipv6 attributes. (cf. #19525)

----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [x] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [x] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-rds/test/cluster.test.ts [hunk 1]
```diff
@@ -457,6 +457,25 @@ describe('cluster', () => {
     }).toThrow(/You cannot specify both parameterGroup and parameters/);
   });
 
+  test('instance with dual-stack network type', () => {
```
Comment (TheRealAmazonKendra) [inline packages/@aws-cdk/aws-rds/test/cluster.test.ts:460]: We need a test case where IPV4 is explicitly set.

File: packages/@aws-cdk/aws-rds/test/instance.test.ts [hunk 2]
```diff
@@ -321,6 +321,20 @@ describe('instance', () => {
     });
   });
 
+  test('instance with dualstack network type', () => {
```
Comment (TheRealAmazonKendra) [inline packages/@aws-cdk/aws-rds/test/instance.test.ts:324]: We need a test case where IPV4 is explicitly set.

File: packages/@aws-cdk/aws-rds/test/integ.cluster-dual.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+const vpc = new ec2.Vpc(stack, 'VPC', { maxAzs: 2, natGateways: 0 });
+const ipv6 = new ec2.CfnVPCCidrBlock(stack, 'Ipv6CidrBlock', { vpcId: vpc.vpcId, amazonProvidedIpv6CidrBlock: true });
+vpc.isolatedSubnets.forEach((subnet, idx) => {
+  const cfnSubnet = subnet.node.defaultChild as ec2.CfnSubnet;
+  cfnSubnet.ipv6CidrBlock = cdk.Fn.select(idx, cdk.Fn.cidr(cdk.Fn.select(0, vpc.vpcIpv6CidrBlocks), 256, '64'));
+  cfnSubnet.addDependsOn(ipv6);
+});
+
+new rds.DatabaseCluster(stack, 'Cluster', {
```
Comment (TheRealAmazonKendra) [inline packages/@aws-cdk/aws-rds/test/integ.cluster-dual.ts:17]: We need a test case where IPV4 is explicitly set.

File: packages/@aws-cdk/aws-rds/test/integ.instance-dual.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+const vpc = new ec2.Vpc(stack, 'VPC', { maxAzs: 2, natGateways: 0 });
+const ipv6 = new ec2.CfnVPCCidrBlock(stack, 'Ipv6CidrBlock', { vpcId: vpc.vpcId, amazonProvidedIpv6CidrBlock: true });
+vpc.isolatedSubnets.forEach((subnet, idx) => {
+  const cfnSubnet = subnet.node.defaultChild as ec2.CfnSubnet;
+  cfnSubnet.ipv6CidrBlock = cdk.Fn.select(idx, cdk.Fn.cidr(cdk.Fn.select(0, vpc.vpcIpv6CidrBlocks), 256, '64'));
+  cfnSubnet.addDependsOn(ipv6);
+});
+
+new rds.DatabaseInstance(stack, 'Instance', {
```
Comment (TheRealAmazonKendra) [inline packages/@aws-cdk/aws-rds/test/integ.instance-dual.ts:17]: We need a test case where IPV4 is explicitly set.

**Review by TheRealAmazonKendra — CHANGES_REQUESTED** [review]
This looks pretty good, we just need the extra tests I commented below.

**Comment by Tietew** [comment]
@TheRealAmazonKendra Thank you for review I've added tests for IPV4 explicitly.

**Pull request merged**